R/mock_httr_response.R

```diff
@@ -0,0 +1,30 @@
+#' Mocks an httr response
+#'
+#' @param status code integer.
+#' @param content list. 
+#' @param type character. json or text
+#' @return a response object.
+#' @export
+#' @examples
+#' mock_httr_response(200L, list(data = "it works!"), "json")
+mock_httr_response <- function(status_code, content, type) {
+  type <- match.arg(type, c("json", "text"))
+  if (identical(type, "json")) {
+    if(is.list(content)) { content <- jsonlite::toJSON(content, auto_unbox = TRUE) }
+    httr:::response(
```

Comment:
`getFromNamespace("response", "httr")` would only be needed once or above every :::response call?

 LIke:
`getFromNamespace("response", "httr")`
`    httr:::response(
      status_code = status_code,
      content = charToRaw(content),
      headers = list(`Content-Type` = "application/json")
    )